syntax = "proto3";

package user;

service AdminService {
  rpc CreateUser (CreateUserRequest) returns (UserResponse);
  rpc GetUser (GetUserRequest) returns (UserResponse);
  rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
  rpc DeleteUser (GetUserRequest) returns (UserDeletedResponse);
}

message CreateUserRequest {
  string email = 1;
  string name = 2;
  string password = 3;
}

message GetUserRequest {
  string id = 1;
}

message UpdateUserRequest {
  string id = 1;
  string name = 2;
  string email = 3;
}

message UserResponse {
  string id = 1;
  string email = 2;
  string name = 3;
  string created_at = 4;
}

message UserDeletedResponse {
  bool success = 1;
}


service UserService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc GetMyProfile (EmptyRequest) returns (UserResponse);
  rpc UpdateMyName (UpdateMyNameRequest) returns (UserResponse);
}

message LoginRequest {
  string email = 1;
  string password = 2;
}
message LoginResponse {
  string token = 1;
}

message EmptyRequest {} // Новый пустой запрос для GetMyProfile


message UpdateMyNameRequest {
  string name = 1;
}

